### PR TITLE
[BEAM-577] Add support for reading compressed files to filebasedsource.

### PR DESCRIPTION
**Pull request by sbilac**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by sbilac** [comment]
R: chamikaramj

**Comment by sbilac** [comment]
R: @chamikaramj

File: sdks/python/apache_beam/io/filebasedsource.py [hunk 1]
```diff
@@ -91,18 +92,45 @@ def default_output_coder(self):
 class FileBasedSource(iobase.BoundedSource):
   """A ``BoundedSource`` for reading a file glob of a given type."""
 
-  def __init__(self, file_pattern, min_bundle_size=0, splittable=True):
+  def __init__(self,
+               file_pattern,
+               min_bundle_size=0,
+               # TODO(slaven): Add support for AUTO once cl/130858239 is in.
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/filebasedsource.py]: # TODO(slaven): Add support for automatically determining the type of compressed files.

Comment (sbilac) [inline sdks/python/apache_beam/io/filebasedsource.py]: Done

**Comment by chamikaramj** [comment]
LGTM

File: sdks/python/apache_beam/io/filebasedsource.py [hunk 2]
```diff
@@ -91,18 +92,46 @@ def default_output_coder(self):
 class FileBasedSource(iobase.BoundedSource):
   """A ``BoundedSource`` for reading a file glob of a given type."""
 
-  def __init__(self, file_pattern, min_bundle_size=0, splittable=True):
+  def __init__(self,
+               file_pattern,
+               min_bundle_size=0,
+               # TODO(slaven): Add support for automatically determining the
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/filebasedsource.py]: Actually, can you please file a JIRA for this and remove this TODO.

**Comment by sbilac** [comment]
R: @dhalperi

File: sdks/python/apache_beam/io/filebasedsource.py (hunk 2 above)
Comment (dhalperi) [inline sdks/python/apache_beam/io/filebasedsource.py]: still needs work

Comment (sbilac) [inline sdks/python/apache_beam/io/filebasedsource.py]: Done

**Comment by dhalperi** [comment]
LGTM, merged. Thanks.

Please close the PR at your leisure.

**Comment by sbilac** [comment]
Thanks.

Slaven

**Pull request closed without merging**